pkgs/desktops/gnome-3/extensions/emoji-selector/default.nix

```diff
@@ -0,0 +1,34 @@
+{ stdenv, fetchFromGitHub, glib, gettext }:
+
+stdenv.mkDerivation rec {
+  pname = "gnome-shell-emoji-selector";
+  version = "19";
+
+  src = fetchFromGitHub {
+    owner = "maoschanz";
+    repo = "emoji-selector-for-gnome";
+    rev = "${version}";
+    sha256 = "0x60pg5nl5d73av494dg29hyfml7fbf2d03wm053vx1q8a3pxbyb";
+  };
+
+  uuid = "emoji-selector@maestroschan.fr";
+
+  buildInputs = [ glib ];
+
+  buildPhase = ''
+    glib-compile-schemas ./${uuid}/schemas
+  '';
+
+  installPhase = ''
+    mkdir -p $out/share/gnome-shell/extensions
```

Comment:
When overriding `installPhase`, you should also provide `runHook preInstall` and `runHook postInstall`.

Reply:
Same for `buildPhase`.